metric_learn/base_metric.py

```diff
@@ -409,8 +409,10 @@ def set_threshold(self, threshold):
       The pairs classifier with the new threshold set.
     """
     check_is_fitted(self, 'preprocessor_')
-
-    self.threshold_ = threshold
+    if threshold is isinstance(threshold, (bool)):
+        raise ValueError('Parameter threshold must be a real number. '
+                         'Got {} instead.'.format(type(threshold)))
+    self.threshold_ = float(threshold)
```

Comment:
Two comments:
- Not sure if it makes sense to make an exception for `bool`
- Maybe it is good to throw the custom error in all cases, which may be more informative for users? what do you think @perimosocordiae?

Reply:
I agree, and the first approach was that, then I changed it to `float(threshold)`. But given that it works for al except `bool`, maybe its better to use the custom error for all cases as proposed initially.